DependencyInjection/MonologExtension.php

```diff
@@ -138,7 +139,12 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
             return $handlerId;
         }
 
-        $definition = new Definition($this->getHandlerClassByType($handler['type']));
+        $handlerClass = $this->getHandlerClassByType($handler['type']);
+        $definition = new Definition($handlerClass);
+        $handlerInterfaces = class_implements($handlerClass);
+        if ($handlerInterfaces && in_array(ResettableInterface::class, $handlerInterfaces, true)) {
```

Comment:
Couldn't we use `is_subclass_of` to check if the class implements the interface?

`if (is_subclass_of($handlerClass, ResettableInterface::class))`